### PR TITLE
refactor authorization for sanity

### PR DESCRIPTION
**Pull request by deads2k**
@liggitt This is a straight "move" refactor.  It splits authorization.go into logical components for better organization.

### CODE REVIEW

File: pkg/authorization/authorizer/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+type AuthorizationAttributes interface {
+	GetVerb() string
+	// GetResource returns the resource type.  If IsNonResourceURL() is true, then GetResource() is "".
+	GetResource() string
+	GetResourceName() string
+	// GetRequestAttributes is of type interface{} because different verbs and different Authorizer/AuthorizationAttributeBuilder pairs may have different contract requirements.
+	GetRequestAttributes() interface{}
+	// IsNonResourceURL returns true if this is not an action performed against the resource API
+	IsNonResourceURL() bool
+	// GetURL returns the URL split on '/'s
```
Comment (liggitt) [inline pkg/authorization/authorizer/interfaces.go]: a string can't be split :) should doc whether it has a leading '/'

Comment (deads2k) [inline pkg/authorization/authorizer/interfaces.go]: updated

**Comment by liggitt** [comment]
LGTM

**Comment by liggitt** [comment]
[merge]

**Pull request merged**